backend/user/notifications.py

```diff
@@ -4,29 +4,41 @@
 from apns2.client import APNsClient
 from apns2.credentials import TokenCredentials
 from apns2.payload import Payload
+from celery import shared_task
 
 
 # taken from the apns2 method for batch notifications
 Notification = collections.namedtuple("Notification", ["token", "payload"])
 
 
+# celery delayed tasks
+def send_delayed_push_notif(token, title, body, countdown, isDev=False):
+    send_push_notif.apply_async(args=[token, title, body, isDev], countdown=countdown)
+
+
+def send_delayed_push_notif_batch(tokens, title, body, countdown, isDev=False):
+    send_push_notif_batch.apply_async(args=[tokens, title, body, isDev], countdown=countdown)
+
+
+@shared_task(name="tasks.send_push_notif")
 def send_push_notif(token, title, body, isDev=False):
     """Sends single push notification to token object"""
 
     client = get_client(isDev)
     alert = {"title": title, "body": body}
     payload = Payload(alert=alert, sound="default", badge=0)
     topic = "org.pennlabs.PennMobile" + (".dev" if isDev else "")
-    client.send_notification(token.token, payload, topic)
+    client.send_notification(token, payload, topic)
 
 
+@shared_task(name="tasks.send_push_notif_batch")
 def send_push_notif_batch(tokens, title, body, isDev=False):
```

Comment:
this code and `send_shadow_push_notif_batch` looks about exactly the same except for one line. might be a good idea to just have 1 method and have a parameter for `send_shadow` or something, and it defaults to `False`